samples/openthread/coap_client/prj.conf

```diff
@@ -15,6 +15,13 @@ CONFIG_COAP_CLIENT_LOG_LEVEL_DBG=y
 CONFIG_COAP_CLIENT_UTILS_LOG_LEVEL_DBG=y
 CONFIG_COAP_UTILS_LOG_LEVEL_DBG=y
 
+# Generic networking options
+CONFIG_NETWORKING=y
+CONFIG_NET_CONNECTION_MANAGER=y
```

Comment:
I believe this can be removed since it's inherited from Zephyr samples but not used in our Thread samples.

Reply:
Not the `NETWORKING` as that is required for the `NET_L2_OPENTHREAD` symbol to be present:
https://github.com/nrfconnect/sdk-zephyr/blob/9807bcf4e21610727f551e0a4f149fe4dc7a4289/subsys/net/Kconfig#L74
https://github.com/nrfconnect/sdk-zephyr/blob/9807bcf4e21610727f551e0a4f149fe4dc7a4289/subsys/net/Kconfig#L85

but maybe `NET_CONNECTION_MANAGER`.
But i'll let @canisLupus1313 followup, as current states is only for showing how things can be cleaned up, and not mergable.